controllers/secretagentconfiguration_controller.go

```diff
@@ -0,0 +1,230 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+	"reflect"
+
+	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	secretagentv1alpha1 "github.com/ForgeRock/secret-agent/api/v1alpha1"
+
+	"github.com/ForgeRock/secret-agent/pkg/generator"
+	"github.com/ForgeRock/secret-agent/pkg/k8ssecrets"
+	"github.com/ForgeRock/secret-agent/pkg/memorystore"
+	"github.com/ForgeRock/secret-agent/pkg/secretsmanager"
+	secretagenttypes "github.com/ForgeRock/secret-agent/pkg/types" //TODO: Part of datatype workaround
+	"github.com/go-playground/validator/v10"
+	"gopkg.in/yaml.v2"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+// SecretAgentConfigurationReconciler reconciles a SecretAgentConfiguration object
+type SecretAgentConfigurationReconciler struct {
+	client.Client
+	Log    logr.Logger
+	Scheme *runtime.Scheme
+}
+
+// +kubebuilder:rbac:groups=secret-agent.secrets.forgerock.io,resources=secretagentconfigurations,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=secret-agent.secrets.forgerock.io,resources=secretagentconfigurations/status,verbs=get;update;patch
+// +kubebuilder:rbac:groups="",resources=secrets,verbs=get;list;watch;create;update;patch;delete
+
+// Reconcile function
+func (r *SecretAgentConfigurationReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
+	ctx := context.Background()
+	log := r.Log.WithValues("secretagentconfiguration", req.NamespacedName)
+
+	// your logic here
+	var instance secretagentv1alpha1.SecretAgentConfiguration
+	if err := r.Get(ctx, req.NamespacedName, &instance); err != nil {
+		log.Error(err, "unable to fetch SecretAgentConfiguration")
+		// we'll ignore not-found errors, since they can't be fixed by an immediate
+		// requeue (we'll need to wait for a new notification), and we can get them
+		// on deleted requests.
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	//Old main.go
+
+	// TODO Datatype workaround
+	data, err := yaml.Marshal(instance.Spec)
+	config := &secretagenttypes.Configuration{}
+	err = yaml.Unmarshal(data, config)
+	//
+
+	//Namespace workaround: Populate the namespace field. Expected by downstream functions.
+	for index := range config.Secrets {
+		config.Secrets[index].Namespace = instance.Namespace
+	}
+
+	//TODO Change to validating webhook: https://book.kubebuilder.io/cronjob-tutorial/webhook-implementation.html
+	validate := validator.New()
+	validate.RegisterStructValidation(secretagentv1alpha1.ConfigurationStructLevelValidator, secretagentv1alpha1.Configuration{})
+	err = validate.Struct(config)
+	if err != nil {
+		log.Error(err, "error validating configuration file: %+v")
```

Comment:
Shouldn't this be a `return ctrl.Result{}, err`?
Same with lines 92, 98, 107, and so on.